lib/chef/knife/vsphere_vm_list.rb

```diff
@@ -47,11 +56,16 @@ def print_vm(vm)
     puts "\t\t#{ui.color('State:', :magenta)} #{state}"
   end
 
+  def print_folder(folder)
+    puts "#{ui.color('Folder', :cyan)}: " + (folder.path[3..-1].map { |x| x[1] }.* '/')
+  end
+
   def run
     vim_connection
     base_folder = find_folder(get_config(:folder))
-    recurse =  get_config(:recursive)
+    only_folders = get_config(:onlyfolders)
+    recurse = only_folders ? true : get_config(:recursive)
```

Comment:
i don't have enough ruby experience to feel like i'm allowed much of an
opinion :)  i want to do what's most kosher with the group.  i'll change it
to ternary.  thanks

On Wed, Sep 2, 2015 at 10:42 AM, Sean Walberg <notifications@github.com>
wrote:

> In lib/chef/knife/vsphere_vm_list.rb
> <https://github.com/adamleff/knife-vsphere/pull/195#discussion_r38539629>:
>
> >    def run
> >      vim_connection
> >      base_folder = find_folder(get_config(:folder))
> > -    recurse =  get_config(:recursive)
> > +    only_folders = get_config(:onlyfolders)
> > +    recurse = only_folders ? true : get_config(:recursive)
>
> Hi, since we're dealing entirely with booleans and want to represent a
> priority, what would you think about using the boolean operators instead of
> the ternary operator?
>
> recurse = only_folders || get_config(:recursive)
>
> The ternary operator is a great way to make an if/else very concise, but
> I find that if we're dealing entirely in booleans that the regular
> operators read more English-like, and are therefore more expressive.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adamleff/knife-vsphere/pull/195/files#r38539629>.
>